src/app-interoperability/application-package-definition.linkml.yaml

```
@@ -0,0 +1,497 @@
+version: 1.0 #Arne: update later
+prefixes:
+  linkml: https://w3id.org/linkml/
+  margo: https://specification.margo.org/
+imports:
+  - linkml:types
+
+default_prefix: margo
+
+
+# Class Definitions
+classes:
+  ApplicationDescription:
+    description: Root class for an application description.
+    attributes:
+      apiVersion:
+        description: Identifier of the version of the API the object definition follows.
+        required: true
+        range: string
+        rank: 10
+      kind:
+        description: Specifies the object type; must be 'application'.
+        range: string
+        required: true
+        pattern: ^application$ #Arne: should this be changed to "ApplicationDescription"?
+        rank: 20
+      metadata:
+        description: >-
+          Metadata element specifying characteristics about the application deployment.
+          See the [Metadata Attributes](#metadata-attributes) section below.
+        range: Metadata
+        required: true
+        rank: 30
+      deploymentProfiles:
+        description: >-
+          Deployment profiles element specifying the types of deployments the application supports. 
+          See the [Deployment](#deployment-profiles-attributes) section below.
+        range: DeploymentProfile #Arne: mistake in spec: called "Deployment" but type is called "DeploymentProfile"
+        multivalued: false
+        required: true
+        rank: 40
+      parameters:
+        description: >-
+          Parameters element specifying the configurable parameters to use when installing, or updating, the application.
+          See the [Parameter](#parameter-attributes)
+        range: Parameter #Arne: mmmh, spec defines this as map[string][Parameter]
+        multivalued: true
+        rank: 50
+      configuration:
+        description: >-
+          Configuration element specifying how parameters should be displayed to the user for setting the value 
+          as well as the rules to use to validate the user's input. 
+          See the [Configuration](#configuration-attributes) section below.
+        range: Configuration
+        rank: 60
+
+  Metadata:
+    description: Metadata about the application.
+    rank: 10
+    attributes:
+      id:
+        description: >-
+          An identifier for the application. The id is used to help create unique identifiers where required, 
+          such as namespaces. The id must be lower case letters and numbers and MAY contain dashes. 
+          Uppercase letters, underscores and periods MUST NOT be used. The id MUST NOT be more than 200 characters.
+        rank: 10
+        range: string
+        required: true
+        pattern: ^[a-z0-9-]{1,200}$
+      name:
+        description: >-
+          The application's official name. 
+          This name is for display purposes only and can container whitespace and special characters.
+        rank: 20
+        range: string
+        required: true
+      version:
```

@arne-broering `description` appears to be missing. Is it required?

```suggestion
      description:
            range: string
      version:
```